x-pack/plugins/security_solution/public/detection_engine/rule_details_ui/pages/rule_details/index.tsx

```diff
@@ -240,9 +240,10 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
   const isLoading = ruleLoading && rule == null;
 
   const { starting: isStartingJobs, startMlJobs } = useStartMlJobs();
-  const startMlJobsIfNeeded = useCallback(async () => {
-    await startMlJobs(rule?.machine_learning_job_id);
-  }, [rule, startMlJobs]);
+  const startMlJobsIfNeeded = useCallback(
+    () => startMlJobs(rule?.machine_learning_job_id),
+    [rule, startMlJobs]
+  );
```

Comment:
So there's going to be an issue with our prebuilt rules and updating the `ml_job_id` as the rules will most likely already be installed before the jobs, and since the rules are immutable, they can't be updated via the API (only server side via the `rulesClient`) until we finish adding support for modifying immutable rules (hopefully in `8.8`).

For instance, here's the `Rule Details` page for the `Unusual Linux Network Activity` prebuilt rule when the corresponding _space-aware_ job is installed:

<p align="center">
  <img width="800" src="https://user-images.githubusercontent.com/2946766/220776943-51f464b8-4234-4aa4-9ab5-3c14af9cad7a.png" />
</p>

<p align="center">
  <img width="500" src="https://user-images.githubusercontent.com/2946766/220778238-a298f814-77a3-4532-9975-56771a6402a7.png" />
</p>

Reply:
Ouch! It looks like a blocker.

Reply:
Would it be possible to move the job creation to the BE for all rules? So, whenever a rule containing a `module_job_id` is created it calls the job creation API and assigns the right `job_id` to the rule.

Reply:
> Would it be possible to move the job creation to the BE for all rules? So, whenever a rule containing a `module_job_id` is created it calls the job creation API and assigns the right `job_id` to the rule.

Yeah, we might want to finally just move our logic to the backend so we have the option to add additional logic like this. This would also allow bypassing the immutability issue since we could update the rule directly from the `rulesClient`.